lib/appcanary/configuration.rb

```diff
@@ -0,0 +1,51 @@
+require "yaml"
+
+module Appcanary
+  APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3/monitors"
+
+  class Configuration
+    attr_accessor :base_uri, :api_token, :monitor_name
+
+    def initialize
+      self.base_uri = APPCANARY_DEFAULT_BASE_URI
+    end
+  end
+
+  class ConfigurationError < RuntimeError
+  end
+
+  class << self
```

Comment:
IF I REMEMBER CORRECTLY,

```ruby
class << self
  def foo
``` 

is the equivalent of writing

```ruby
def self.foo
```

there is some minor tradeoff that has since caused it be less popular in the idiom I think? but I cannot enumerate it to you.